File: app/src/main/java/igrek/songbook/activity/CurrentActivityListener.java
package igrek.songbook.activity;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

public class CurrentActivityListener implements Application.ActivityLifecycleCallbacks {
	
	private Activity currentActivity;
	
	public Activity getCurrentActivity() {
		return currentActivity;
	}
	
	@Override
	public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
		currentActivity = activity;
	}
	
	@Override
	public void onActivityStarted(Activity activity) {
		currentActivity = activity;
	}
	
	@Override
	public void onActivityResumed(Activity activity) {
		currentActivity = activity;
	}
	
	@Override
	public void onActivityPaused(Activity activity) {

	}
	
	@Override
	public void onActivityStopped(Activity activity) {

	}
	
	@Override
	public void onActivitySaveInstanceState(Activity activity, Bundle outState) {

	}
	
	@Override
	public void onActivityDestroyed(Activity activity) {

	}
}

File: app/src/main/java/igrek/songbook/activity/CurrentActivityListener.kt
package igrek.songbook.activity

import android.app.Activity
import android.app.Application
import android.os.Bundle

class CurrentActivityListener : Application.ActivityLifecycleCallbacks {

    var currentActivity: Activity? = null
        private set

    override fun onActivityCreated(activity: Activity?, savedInstanceState: Bundle?) {
        currentActivity = activity
    }

    override fun onActivityStarted(activity: Activity?) {
        currentActivity = activity
    }

    override fun onActivityResumed(activity: Activity?) {
        currentActivity = activity
    }

    override fun onActivityPaused(activity: Activity?) {
    }

    override fun onActivityStopped(activity: Activity?) {
    }

    override fun onActivitySaveInstanceState(activity: Activity?, outState: Bundle?) {
    }

    override fun onActivityDestroyed(activity: Activity?) {
    }
}
